core/app/assets/javascripts/card_formatting.js

```diff
@@ -0,0 +1,46 @@
+/* global Cleave */
+
+/* eslint-disable no-unused-vars */
+function formatAllCardInputFields () {
+  formatCardnumber()
+  formatCardExpiry()
+  formatCardCode()
+}
+
+function formatCardnumber () {
+  if (document.querySelector('.cardNumber')) {
+    document.querySelectorAll('.cardNumber').forEach(function (cardNumber) {
+      /* eslint-disable no-new */
+      new Cleave(cardNumber, {
+        creditCard: true,
+        onCreditCardTypeChanged: function (type) {
+          $('.ccType').val(type)
```

Comment:
@MatthewKennedy shouldn't this be scoped to match the ccType for this particular card?

Reply:
When you start typing and a recognised card type is found, that code adds the card type to this hidden input:
<img width="730" alt="Screenshot 2021-02-15 at 08 16 12" src="https://user-images.githubusercontent.com/1240766/107921291-3915a800-6f66-11eb-80b1-f8dff915205c.png">

Do you want it to also add a graphical representation of the card type at the end on the card number input field as below:
<img width="609" alt="Screenshot 2021-02-15 at 08 19 43" src="https://user-images.githubusercontent.com/1240766/107921563-9b6ea880-6f66-11eb-87ed-90f8b2ca8439.png">

Reply:
I'll check the scope, one moment.